### PR TITLE
[v1.0] Add custom FlexVol plugin directory support

### PR DESCRIPTION
**Pull request by Oats87**
Adding ability to recognize network plugin specific entry for flexvolume directory

https://github.com/rancher/rke/issues/1744

### CODE REVIEW

**Comment by Oats87** [comment]
> Is there a reason why this can't go in `defaults.go`?

No, I'll move it. Thanks

**Comment by Oats87** [comment]
@superseb I moved the defaults to `defaults.go`

File: cluster/defaults.go [hunk 1]
```diff
@@ -64,6 +64,10 @@ const (
 	DefaultFlannelBackendVxLanPort = "8472"
 	DefaultFlannelBackendVxLanVNI  = "1"
 
+	DefaultCalicoFlexVolPluginDirectory = "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds"
```
Comment (superseb) [inline cluster/defaults.go]: Do we expect these to be different in the future as we are setting two of the same strings as default?

Comment (Oats87) [reply cluster/defaults.go]: These were kept separate for future-proofing in the event that we need to change it specifically for a plugin without having to change the rest of `defaults.go`

**Comment by Oats87** [comment]
> What is considered regarding this change? This is dependent on kubelet `volume-plugin-dir` and kube-controller-manager `flex-volume-plugin-dir`. Was it considered to base it of the configuration or is it not strictly dependent on that? Does it not need to be validated in any case, is it going to be documented to change what is needed to configure it, or are we only providing a way to move it off the default location?

Generally, when someone is changing the `flex-volume-plugin-dir` they are doing some pretty advanced things on their cluster. I am thinking that we can simply document our way around this by specifying the multiple places you have to specify the `flex-volume-plugin-dir` for both the `kubelet` and `kube-controller-manager` similar to the way we handle service IP CIDR changes today.

**Comment by Oats87** [comment]
> What is considered regarding this change? This is dependent on kubelet `volume-plugin-dir` and kube-controller-manager `flex-volume-plugin-dir`. Was it considered to base it of the configuration or is it not strictly dependent on that? Does it not need to be validated in any case, is it going to be documented to change what is needed to configure it, or are we only providing a way to move it off the default location?

Generally, when someone is changing the `flex-volume-plugin-dir` they are doing some pretty advanced things on their cluster. I am thinking that we can simply document our way around this by specifying the multiple places you have to specify the `flex-volume-plugin-dir` for both the `kubelet` and `kube-controller-manager` similar to the way we handle service IP CIDR changes today.

**Pull request merged**